### PR TITLE
TNO-2326 Parse tags

### PR DESCRIPTION
**Pull request by Fosol**
Fixed issue where tags were not being parsed from the transcription tab.

image

### CODE REVIEW

File: app/editor/src/features/content/form/ContentForm.tsx [hunk 1]
```diff
@@ -125,10 +125,6 @@ const ContentForm: React.FC<IContentFormProps> = ({
   const source = sources.find((s) => s.id === form.sourceId);
   const program = series.find((s) => s.id === form.seriesId);
 
-  const parseTags = (x: string[]) => {
```
Comment (Fosol) [inline app/editor/src/features/content/form/ContentForm.tsx:128]: Not needed

File: app/editor/src/features/content/form/ContentForm.tsx [hunk 2]
```diff
@@ -546,7 +542,7 @@ const ContentForm: React.FC<IContentFormProps> = ({
                   <Show visible={props.values.contentType === ContentTypeName.Image}>
                     <ContentStoryForm
                       contentType={ContentTypeName.Image}
-                      checkTags={(c: string[]) => parseTags(c)}
+                      setParsedTags={setParsedTags}
```
Comment (Fosol) [inline app/editor/src/features/content/form/ContentForm.tsx:545]: Named based on what it does

File: app/editor/src/features/content/form/ContentStoryForm.tsx [hunk 3]
```diff
@@ -21,7 +22,7 @@ import { isSummaryRequired } from './utils';
 export interface IContentStoryFormProps {
   contentType: ContentTypeName;
   summaryRequired?: boolean;
-  checkTags?: Function;
+  setParsedTags?: React.Dispatch<React.SetStateAction<string[]>>;
```
Comment (Fosol) [inline app/editor/src/features/content/form/ContentStoryForm.tsx:25]: Using typescript to specify what something is to stop bugs

File: app/editor/src/features/content/form/ContentStoryForm.tsx [hunk 4]
```diff
@@ -32,39 +33,18 @@ export interface IContentStoryFormProps {
...
   const [showExpandModal, setShowExpandModal] = React.useState(false);
   const { height } = useWindowSize();
   const [{ tags }] = useLookup();
+  const getTags = useExtractTags({ setParsedTags });
 
   const [summaryRequired, setSummaryRequired] = React.useState(
     initSummaryRequired ?? isSummaryRequired(values),
   );
 
-  const getTags = (name: string, e: string) => {
```
Comment (Fosol) [inline app/editor/src/features/content/form/ContentStoryForm.tsx:46]: Moved to hook to share with other component

File: app/editor/src/features/content/form/ContentStoryForm.tsx [hunk 5]
```diff
@@ -95,7 +75,7 @@ export const ContentStoryForm: React.FC<IContentStoryFormProps> = ({
           name="body"
           expandModal={setShowExpandModal}
           tags={tags}
-          onChange={(e) => getTags('body', e)}
+          onChange={(text) => getTags('body', text)}
```
Comment (Fosol) [inline app/editor/src/features/content/form/ContentStoryForm.tsx:78]: Renamed to be what it is

File: app/editor/src/features/content/form/hooks/useExtractTags.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { useFormikContext } from 'formik';
+import { useLookup } from 'store/hooks';
+
+import { IContentForm } from '../interfaces';
+
+export interface IExtractTagsProps {
+  setParsedTags?: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+export const useExtractTags = ({ setParsedTags }: IExtractTagsProps) => {
```
Comment (Fosol) [inline app/editor/src/features/content/form/hooks/useExtractTags.ts:10]: Separate hook to reuse on both components, I'd like to extract the passed in function too.

**Review by kylgarmor — APPROVED** [review]

**Pull request merged**